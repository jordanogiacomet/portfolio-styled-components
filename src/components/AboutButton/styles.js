import styled, { css } from 'styled-components';

export const Container = styled.button`
  background: transparent;
  position: relative;
  border: none;
  color: #ADB7BE;
  margin-right: 12px;
  cursor: pointer;
  padding: 0.5rem 1rem;
  &.light{
    color: black;
    ${({ active }) =>
    active &&
    css`
      color: #1E90FF;
      &::after{
        width: 100%;
      }
    `}
  }
  &:hover {
    color: #1E90FF;
  }
  &::after {
    content: '';
    display: block;
    height: 2px;
    background: linear-gradient(to right, #1E90FF, #4169E1, #00008B);
    transition: width 0.3s ease-in-out;
    width: 0;
    position: absolute;
    left: 0;
    bottom: 0;
  }
  &:hover::after {
    width: 100%;
  }
  @media (min-width: 768px) {
    font-size: 2rem;
  }
  @media (max-width: 640px) {
    font-size: 1.6rem;
  }
  ${({ active }) =>
    active &&
    css`
      color: #1E90FF;
      &::after{
        width: 100%;
      }
    `}
`;
