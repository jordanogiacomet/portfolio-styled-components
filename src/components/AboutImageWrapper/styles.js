import styled from 'styled-components';

export const Container = styled.div`
  margin-top: 4px;
  order: 2;
  > img {
  display: block;
  max-width: 100%;
  height: auto;
  border-radius: 10px; /* Ajuste o raio da borda conforme necess√°rio */
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1), 0 6px 20px rgba(0, 0, 0, 0.1);
  }
  
  @media (min-width: 768px) {
  margin-top: 0;
  }
`;
