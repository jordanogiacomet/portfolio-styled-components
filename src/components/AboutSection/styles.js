import styled from 'styled-components';

export const Container = styled.section`
  color: white;
  padding: 50px 0px;
  display: flex;
  flex-direction: column;
  @media (min-width: 768px) {
    display: grid;
    grid-template-columns: 1fr 1fr;
    gap: 50px;
    align-items: center;
    margin-top: 80px;
  }
  @media (min-width: 1280px) {
    gap: 50px;

  };
`;
