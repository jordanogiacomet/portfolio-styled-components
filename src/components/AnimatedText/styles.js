import styled from 'styled-components';
import { TypeAnimation } from 'react-type-animation';

export const Container = styled(TypeAnimation)`
  display: inline-block;
  
  &.light{
    color: ${({ theme }) => theme.colors.primaryColor}
  }
  &.dark{
    ${({ theme }) => theme.colors.secondaryColor}
  }
`;