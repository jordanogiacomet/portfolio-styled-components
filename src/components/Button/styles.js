import styled, { css } from 'styled-components';

const getButtonStyles = ({ theme, type }) => {
  const buttonType = theme.button[type] || theme.button.default;
  return css`
    padding: ${buttonType.padding};
    border-radius: ${buttonType.borderRadius};
    color: ${buttonType.textColor};
    background: ${buttonType.background};
    font-weight: bold;
    cursor: pointer; 
    display: inline-flex;
    align-items: center;
    justify-content: center;
    min-width: 150px; /* Ajuste a largura mínima conforme necessário */
    transition: scale ease-in-out 0.2s;
    &:hover {
      transition: scale ease-in-out 0.2s;
      scale: calc(1.03);
    };

    @media (${theme.media.lteMedium}) {
      width: 100%; /* Largura 100% em telas menores */
    }
  `;
};

export const Container = styled.button`
  ${getButtonStyles};
`; 