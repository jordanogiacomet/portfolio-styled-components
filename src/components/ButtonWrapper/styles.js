import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  gap: 1rem;
  flex-wrap: wrap;
  margin-top: 2rem;

  @media (max-width: 768px) {
    flex-direction: column;
  }
`;