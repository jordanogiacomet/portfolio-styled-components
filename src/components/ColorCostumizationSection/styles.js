// src/components/ColorCustomizationSection/styles.js
import styled from 'styled-components';

export const Container = styled.section`
  display: flex;
  flex-direction: column;
  gap: 50px;
  margin-bottom: 50px;
`;
