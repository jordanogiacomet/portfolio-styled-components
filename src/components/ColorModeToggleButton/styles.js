import styled from 'styled-components';

export const Container = styled.button`
  padding: 1rem;
  background-color: transparent;
  box-shadow: none;
  border: 2px solid #ADB7BE;
  border-radius: 9999px;
  color: #ADB7BE;
  font-size: 1.2rem; 
  right: 10rem;
  cursor: pointer;
  z-index: 199999;
  transition: 0.2s;
  &:hover{
    color: #4169E1;
    border-color: #4169E1;
  }
  @media(max-width: 768px) {
    bottom: 20px;
  }
  &.light{
    color: white;
    border-color: white;
    &:hover{
      color: #CCCCCC;
      border-color: #CCCCCC;
    }
  }

`;

