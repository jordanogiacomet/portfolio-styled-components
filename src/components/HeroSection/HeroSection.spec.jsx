import { renderTheme } from '../../styles/render-theme';
import { HeroSection } from '.';
import { screen } from '@testing-library/react';

describe('<HeroSection />', () => {
  it('should render the HeroSection with all components', () => {
    renderTheme(<HeroSection />);
    
    expect(screen.getByText("Hello, I'm")).toBeInTheDocument();
    expect(screen.getByText('Hire Me')).toBeInTheDocument();
    expect(screen.getByText('Download CV')).toBeInTheDocument();
    expect(screen.getByAltText('hero image')).toBeInTheDocument();
    expect(screen.getByText('Lorem ipsum dolor sit amet consectetur adipisicing elit. Amet optio autem eveniet molestias, obcaecati nam ab numquam blanditiis ipsam minima odio quidem ea accusantium minus similique deleniti itaque rem. Ab!Lorem ipsum dolor sit amet consectetur adipisicing elit. Amet optio autem eveniet molestias, obcaecati nam ab numquam blanditiis ipsam minima odio quidem ea accusantium minus similique deleniti itaque rem. Ab!Lorem ipsum dolor sit amet consectetur adipisicing elit. Amet optio autem eveniet molestias, obcaecati nam ab numquam blanditiis ipsam minima odio quidem ea accusantium minus similique deleniti itaque rem. Ab!')).toBeInTheDocument();
  });
  it('should match snapshot', () => {
    const { container } = renderTheme(<HeroSection />);

    expect(container).toMatchSnapshot();
  })
});