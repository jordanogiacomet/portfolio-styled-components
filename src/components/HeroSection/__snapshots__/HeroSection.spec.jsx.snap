// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeroSection /> should match snapshot 1`] = `
.c0 {
  display: grid;
  grid-template-columns: repeat(1, 1fr);
}

.c1 {
  grid-column: span 1;
}

.c1.self-center {
  place-self: center;
}

.c1.text-center {
  text-align: center;
}

.c1.mt-4 {
  margin-top: 5rem;
}

.c10 {
  grid-column: span 1;
}

.c10.self-center {
  place-self: center;
}

.c10.text-center {
  text-align: center;
}

.c10.mt-4 {
  margin-top: 5rem;
}

.c4 {
  display: inline-block;
}

.c2 {
  color: white;
  margin-bottom: 1rem;
  font-weight: 800;
  font-size: 2.5rem;
}

.c5 {
  font-size: 1.5rem;
  color: #FFFFFF;
  margin-bottom: 20px;
}

.c3 {
  background: linear-gradient(to right, #a855f7, #ec4899);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}

.c7 {
  padding: 1rem 1.5rem;
  border-radius: 9999px;
  color: #FFFFFF;
  background: linear-gradient(to bottom right, blue, purple, pink);
  cursor: pointer;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  min-width: 150px;
}

.c7:hover {
  background-color: slategray;
}

.c8 {
  padding: 0.4rem;
  border-radius: 9999px;
  color: #FFFFFF;
  background: linear-gradient(to bottom right, blue, purple, pink);
  cursor: pointer;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  min-width: 150px;
}

.c8:hover {
  background-color: darkslategray;
}

.c9 {
  display: block;
  background-color: #121212;
  border-radius: 9999px;
  padding: 1rem 3rem;
}

.c9:hover {
  background-color: darkslategray;
}

.c11 {
  border-radius: 9999px;
  background-color: #181818;
  width: 250px;
  height: 250px;
  position: relative;
}

.c11 img {
  position: absolute;
  transform: translate(-50%, -50%);
  top: 50%;
  left: 50%;
  width: 100%;
  height: 100%;
  object-fit: cover;
  border-radius: 9999px;
}

.c6 {
  display: flex;
  gap: 1rem;
  flex-wrap: wrap;
}

@media (min-width: 640px) {
  .c0 {
    grid-template-columns: repeat(12, 1fr);
    align-items: center;
    gap: 5rem;
  }
}

@media (min-width: 640px) {
  .c1 {
    grid-column: span 7;
  }
}

@media (min-width: 640px) {
  .c1 {
    text-align: left;
  }
}

@media (min-width: 1024px) {
  .c1.mt-4 {
    margin-top: 0;
  }
}

@media (min-width: 640px) {
  .c10 {
    grid-column: span 5;
  }
}

@media (min-width: 640px) {
  .c10 {
    text-align: left;
  }
}

@media (min-width: 1024px) {
  .c10.mt-4 {
    margin-top: 0;
  }
}

@media (min-width: 640px) {
  .c2 {
    font-size: 3rem;
  }
}

@media (min-width: 1024px) {
  .c2 {
    font-size: 4rem;
  }
}

@media (min-width: 640px) {
  .c5 {
    font-size: 1.8rem;
  }
}

@media (min-width: 1024px) {
  .c5 {
    font-size: 2rem;
  }
}

@media ((max-width: 640px)) {
  .c7 {
    width: 100%;
  }
}

@media ((max-width: 640px)) {
  .c8 {
    width: 100%;
  }
}

@media ((max-width: 640px)) {
  .c9 {
    width: 100%;
  }
}

@media (min-width: 1024px) {
  .c11 {
    width: 400px;
    height: 400px;
  }
}

<div>
  <section>
    <div
      class="c0"
    >
      <div
        class="c1"
        span="7"
      >
        <h1
          class="c2"
        >
          <span
            class="c3"
          >
            Hello, I'm
             
          </span>
          <span
            class="index-module_type__E-SaG c4 animated-text"
          />
        </h1>
        <p
          class="c5"
        >
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Amet optio autem eveniet molestias, obcaecati nam ab numquam blanditiis ipsam minima odio quidem ea accusantium minus similique deleniti itaque rem. Ab!Lorem ipsum dolor sit amet consectetur adipisicing elit. Amet optio autem eveniet molestias, obcaecati nam ab numquam blanditiis ipsam minima odio quidem ea accusantium minus similique deleniti itaque rem. Ab!Lorem ipsum dolor sit amet consectetur adipisicing elit. Amet optio autem eveniet molestias, obcaecati nam ab numquam blanditiis ipsam minima odio quidem ea accusantium minus similique deleniti itaque rem. Ab!
        </p>
        <div
          class="c6"
        >
          <button
            class="c7"
            type="hireMe"
          >
            Hire Me
          </button>
          <button
            class="c8"
            type="downloadCV"
          >
            <span
              class="c9"
              type="downloadCV"
            >
              Download CV
            </span>
          </button>
        </div>
      </div>
      <div
        class="c10 self-center mt-4"
        span="5"
      >
        <div
          class="c11"
        >
          <img
            alt="hero image"
            src="/images/hero-image.png"
          />
        </div>
      </div>
    </div>
  </section>
</div>
`;
